plugins/inputs/snmp_trap/snmp_trap_test.go

```diff
@@ -1586,127 +1225,232 @@ func TestReceiveTrapMultipleConfig(t *testing.T) {
-				SecLevel:     tt.secLevel,
-				AuthProtocol: tt.authProto,
-				AuthPassword: config.NewSecret([]byte(tt.authPass + "1")),
-				PrivProtocol: tt.privProto,
-				PrivPassword: config.NewSecret([]byte(tt.privPass + "1")),
-				Translator:   "netsnmp",
+			plugin := &SnmpTrap{
+				ServiceAddress: "udp://:" + strconv.Itoa(port),
+				Version:        "3",
+				Translator:     "netsnmp",
+				SecName:        config.NewSecret([]byte(tt.secName)),
+				SecLevel:       tt.secLevel,
+				AuthProtocol:   tt.authProto,
+				AuthPassword:   config.NewSecret([]byte(tt.authPass)),
+				PrivProtocol:   tt.privProto,
+				PrivPassword:   config.NewSecret([]byte(tt.privPass)),
+				Log:            testutil.Logger{},
+				timeFunc:       time.Now,
 			}
+			require.NoError(t, plugin.Init())
+
+			// inject test translator
+			plugin.transl = &testTranslator{entries: tt.entries}
 
-			// Hook into the trap handler so the test knows when the
-			// trap has been received
-			received2 := make(chan int)
-			wrap2 := func(f gosnmp.TrapHandlerFunc) gosnmp.TrapHandlerFunc {
-				return func(p *gosnmp.SnmpPacket, a *net.UDPAddr) {
-					f(p, a)
-					received2 <- 0
-				}
+			var acc testutil.Accumulator
+			require.NoError(t, plugin.Start(&acc))
+			defer plugin.Stop()
+
+			// Create a v3 client and send the trap
+			var msgFlags gosnmp.SnmpV3MsgFlags
+			switch strings.ToLower(tt.secLevel) {
+			case "noauthnopriv", "":
+				msgFlags = gosnmp.NoAuthNoPriv
+			case "authnopriv":
+				msgFlags = gosnmp.AuthNoPriv
+			case "authpriv":
+				msgFlags = gosnmp.AuthPriv
+			default:
+				require.FailNowf(t, "unknown security level %q", tt.secLevel)
 			}
+			security := createSecurityParameters(tt.authProto, tt.privProto, tt.secName, tt.privPass, tt.authPass)
 
-			// Set up a second service input plugin
-			s2 := &SnmpTrap{
-				ServiceAddress:     "udp://:" + strconv.Itoa(port2),
-				makeHandlerWrapper: wrap2,
-				timeFunc: func() time.Time {
-					return fakeTime
-				},
-				// if cold start be answer otherwise err
-				Log:          testutil.Logger{},
-				Version:      tt.version.String(),
-				SecName:      config.NewSecret([]byte(tt.secName + "2")),
-				SecLevel:     tt.secLevel,
-				AuthProtocol: tt.authProto,
-				AuthPassword: config.NewSecret([]byte(tt.authPass + "2")),
-				PrivProtocol: tt.privProto,
-				PrivPassword: config.NewSecret([]byte(tt.privPass + "2")),
-				Translator:   "netsnmp",
+			client := &gosnmp.GoSNMP{
+				Port:               port,
+				Version:            gosnmp.Version3,
+				Timeout:            2 * time.Second,
+				Retries:            1,
+				MaxOids:            gosnmp.MaxOids,
+				Target:             "127.0.0.1",
+				SecurityParameters: security,
+				SecurityModel:      gosnmp.UserSecurityModel,
+				MsgFlags:           msgFlags,
+				ContextName:        tt.contextName,
+				ContextEngineID:    tt.engineID,
 			}
+			require.NoError(t, client.Connect(), "connecting failed")
+			defer client.Conn.Close()
```

Comment:
as mentioned in the previous comment, the connection is closed twice.

Reply:
See above.